Find a functionally equivalent code:android.widget.FastScroller.scrollTo:COMMENT
Method Modifier: private     
Comment:/**
 * Scrolls to a specific position within the section
 * @param position
 */

@@ -87,95 +87,96 @@
     }
     if (mCurrentSection != sectionIndex) {
         mCurrentSection = sectionIndex;
-        if (transitionPreviewLayout(sectionIndex)) {
+        final boolean hasPreview = transitionPreviewLayout(sectionIndex);
+        if (!mShowingPreview && hasPreview) {
             transitionToDragging();
-        } else {
+        } else if (mShowingPreview && !hasPreview) {
             transitionToVisible();
         }
     }

